Access review v2 api (#9935)
microsoftgraph/microsoft-graph-docs: api-reference/beta/api/accessreviewinstance-listpendingapproval.md

> ---

Reviewer: I think the name of this file should be: accessreviewinstance-pendingaccessreviewinstances.md

> title: "List accessReviewInstances pending approval"

Reviewer: accessReviewInstance: pendingAccessReviewInstances

description: "Retrieve accessReviewInstance objects pending approval by calling user."
localization_priority: Normal
author: "isabelleatmsft"
ms.prod: "microsoft-identity-platform"
doc_type: apiPageType
---

> # List accessReviewInstances pending approval

Reviewer: accessReviewInstance: pendingAccessReviewInstances

Namespace: microsoft.graph

[!INCLUDE [beta-disclaimer](../../includes/beta-disclaimer.md)]

> Retrieve the [accessReviewInstance](../resources/accessreviewinstance.md) objects pending approval by the calling user. A list of zero or more **accessReviewInstance** objects are returned, of which the calling user is an assigned reviewer.

Reviewer: ```suggestion
Retrieve the [accessReviewInstance](../resources/accessreviewinstance.md) objects pending approval by the calling user. A list of zero or more accessReviewInstance objects are returned, of which the calling user is an assigned reviewer.
```

>[!NOTE]
>If many **accessReviewInstances** are returned, to improve efficiency and avoid timeouts, retrieve the result set in pages, by including both the $top query parameter with a page size of at most 100, and the $skip=0 query parameter in the request. When a result set spans multiple pages, Microsoft Graph returns that page with an @odata.nextLink property in the response that contains a URL to the next page of results. If that property is present, continue making additional requests with the @odata.nextLink URL in each response, until all the results are returned, as described in paging Microsoft Graph data in your app.
>
>If no query parameters are provided and there are more than 100 results, Microsoft Graph will automatically paginate results at 100 results per page.

## Permissions

> One of the following permissions is required to call this API. Delegated permissions to personal Microsoft accounts are not supported. To learn more, including how to choose permissions, see [Permissions](/graph/permissions-reference).

Reviewer: ```suggestion
One of the following permissions is required to call this API. To learn more, including how to choose permissions, see [Permissions](/graph/permissions-reference).
```

|Permission type                        | Permissions (from least to most privileged)              |
|:--------------------------------------|:---------------------------------------------------------|

> |Delegated (work or school account)     | AccessReview.Read.All or AccessReview.ReadWrite.All  |

Reviewer: ```suggestion
|Delegated (work or school account)     | AccessReview.Read.All, AccessReview.ReadWrite.All  |
```

> |Application                            | AccessReview.Read.All or AccessReview.ReadWrite.All |

Reviewer: ```suggestion
|Application                            | AccessReview.Read.All, AccessReview.ReadWrite.All |
```

>  The signed in user must also be in a directory role that permits them to read an access review. The signed in user will also only see instances of which they are assigned reviewer in that instance's `accessReviewScheduleDefinition`.

Reviewer: ```suggestion
 The signed-in user must also be in a directory role that permits them to read an access review. The signed in user only sees instances of which they are assigned reviewer in the accessReviewScheduleDefinition of the instance.
```

## HTTP request
<!-- { "blockType": "ignored" } -->
```http

> GET /me/pendingAccessReviewInstances?$expand=definition&$top={pagesize}&$skip=0

Reviewer: ```suggestion
GET /me/pendingAccessReviewInstances
```

Author: The current api documentation has the parameter use here, and I want to be consistent with that. I also want it to be clear that these parameters are necessary.

```
## Request headers
None.

## Request body
Do not supply a request body.

## Response
If successful, this method returns a `200 OK` response code and an array of [accessReviewInstance](../resources/accessreviewinstance.md) objects in the response body.

## Examples

> ##### Request

Reviewer: This should be a H3, not an H4: ### Response

The following example shows a request to retrieve all the access review series in a tenant.

> # [HTTP](#tab/http)

Reviewer: Remove line 51

<!-- {
  "blockType": "request",
  "name": "list_accessReviewInstance_pendingapproval"
}-->
```msgraph-interactive
GET https://graph.microsoft.com/beta/me/pendingAccessReviewInstances?$expand=definition&$top=100&$skip=0
```

> # [C#](#tab/csharp)

Reviewer: Remove lines 60 through 67.

[!INCLUDE [sdk-documentation](../includes/snippets/snippets-sdk-documentation-link.md)]

# [JavaScript](#tab/javascript)
[!INCLUDE [sdk-documentation](../includes/snippets/snippets-sdk-documentation-link.md)]

# [Objective-C](#tab/objc)
[!INCLUDE [sdk-documentation](../includes/snippets/snippets-sdk-documentation-link.md)]

> ##### Response

Reviewer: This should be a H3, not an H4: ### Response
